### ISSUE TITLE
Difficulties reproducing the results

### ISSUE THREAD

**Issue by NamelessGaki** [liangchen527/SLADD#12]
Hello,

Thank you for your exciting work. I have been trying to reproduce the results of the method, but I was unsuccessful. I have run it on a single GPU because of my limited setup and followed the instructions according to the readme.  But after 30 epochs the results are still bad, do you have an idea to why this is the case?
Testing from FF-DF ... Len of data_loader= 219                                                                                                                                                                                                                                                                                                                                                   Acc 0.4821 (~0.00)     |AUC 0.4614 (~0.10)     |EER 0.5286 (~0.10)     |AP 0.5142 (~0.11)                                                                                                 
Testing from FF-NT ...                                                                                                                                                                                        Len of data_loader= 219                                                                                                                                                                                                                                                                                                                                                                                
|Test Loss 14.85385it/s]| Acc 0.4847 (~0.00)     |AUC 0.4335 (~0.11)     |EER 0.5360 (~0.11)     |AP 0.5005 (~0.11)                                                                                                 
[3/4]Testing from FF-FS ...                                                                                                                                                                                        
Len of data_loader= 219                                                                                                                                                                                            |Test Loss 1.436065it/s]|Acc 0.9536 (~0.00)     |AUC 0.9916 (~0.01)     |EER 0.0077 (~0.03)     |AP 0.9975 (~0.01)                                                                                                 
[4/4]Testing from FF-F2F ...                                                                                                                                                                                       
Len of data_loader= 219                                                                                                                                                                                            
219it [00:19, 11.46it/s]                                                                                                                                                                                           
|Test Loss 9.582998it/s]|Acc 0.6704 (~0.00)     |AUC 0.6656 (~0.11)     |EER 0.3854 (~0.13)     |AP 0.7459 (~0.10)

**Comment by chenjunlinlin** [comment]
what is your trainset?, I got another result. like this:
[1/4]Testing from FF-DF ...
|Test Loss 5.7040███████|Acc 0.8024 (~0.00)█████|AUC 0.9025 (~0.06)█████|EER 0.1785 (~0.10)█████|AP 0.9052 (~0.07)

**Comment by ZengqiangChen** [comment]
What is your best_acc after each epoch?

**Comment by Amira-AlSamawi** [comment]
@NamelessGaki 
Could you please tell me what your training batch size is.

I couldn't train on 32 batch size and had to go with 16, but also didn't get good results. I tried halving the learning rate but that didn't work either.

**Comment by ZengqiangChen** [comment]
> @NamelessGaki What is your training batch size?
> 
> I couldn't train on 32 batch size and had to go with 16, but also didn't get good results. I tried halving the learning rate but that didn't work either.

My batch_size is also 16. After each epoch, what is your acc?

**Comment by Amira-AlSamawi** [comment]
I didn't keep tracking of accuracy in each epoch, I cared more about AUC, especially for celebdf dataset.

**Comment by Pudge-tao** [comment]
anyone could provide pretrained model weight? thanks a lot

**Comment by ZengqiangChen** [comment]
> Hello,
> 
> Thank you for your exciting work. I have been trying to reproduce the results of the method, but I was unsuccessful. I have run it on a single GPU because of my limited setup and followed the instructions according to the readme. But after 30 epochs the results are still bad, do you have an idea to why this is the case? Testing from FF-DF ... Len of data_loader= 219 Acc 0.4821 (~0.00) |AUC 0.4614 (~0.10) |EER 0.5286 (~0.10) |AP 0.5142 (~0.11) Testing from FF-NT ... Len of data_loader= 219 |Test Loss 14.85385it/s]| Acc 0.4847 (~0.00) |AUC 0.4335 (~0.11) |EER 0.5360 (~0.11) |AP 0.5005 (~0.11) [3/4]Testing from FF-FS ... Len of data_loader= 219 |Test Loss 1.436065it/s]|Acc 0.9536 (~0.00) |AUC 0.9916 (~0.01) |EER 0.0077 (~0.03) |AP 0.9975 (~0.01) [4/4]Testing from FF-F2F ... Len of data_loader= 219 219it [00:19, 11.46it/s] |Test Loss 9.582998it/s]|Acc 0.6704 (~0.00) |AUC 0.6656 (~0.11) |EER 0.3854 (~0.13) |AP 0.7459 (~0.10)

How long does it take you to train an Epoch? thank you

**Comment by ZengqiangChen** [comment]
> @NamelessGaki Could you please tell me what your training batch size is.
> 
> I couldn't train on 32 batch size and had to go with 16, but also didn't get good results. I tried halving the learning rate but that didn't work either.

How long does it take you to train an Epoch? thank you

**Comment by Ablott123** [comment]
Hi, do you have his test code?

**Comment by ZengqiangChen** [comment]
no





------------------ Original ------------------
From: Ablott123 ***@***.***&gt;
Date: Thu,May 23,2024 1:34 PM
To: liangchen527/SLADD ***@***.***&gt;
Cc: ZengqiangChen ***@***.***&gt;, Comment ***@***.***&gt;
Subject: Re: [liangchen527/SLADD] Difficulties reproducing the results (Issue#12)





 
Hi, do you have his test code?
